File: Assignment9/src/Repo/MemberRepo.js
const load = async (auth, id) => {
  const url = 'https://cse118.com/api/v2/member';
  const headers = {
    Accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: `Bearer ${auth}`,
  };

  const response = await fetch(url, {
    method: 'GET',
    headers: headers,
  });

  const json = await response.json();
  // console.log('Members')
  //console.log(json)
  return json;
};

export {load};

//const get = async (auth, id) => {
//  const url = `https://cse118.com/api/v2/workspace/${id}/member`;
//  const headers = {
//    Accept: 'application/json',
//    Authorization: `Bearer ${auth}`,
//  };
//
//  const response = await fetch(url, {
//    method: 'GET',
//    headers: headers,
//  });
//
//  const json = await response.json();
//  return json;
//};
//
//export { get };
//
//
//const deleteMember = async (auth, wid, mid) => {
//  const urlString = `https://cse118.com/api/v2/workspace/${wid}/member`;
//  const url = new URL(urlString);
//
//  const params = { mid: mid };
//  url.search = new URLSearchParams(params).toString();
//
//  const headers = {
//    Accept: 'application/json',
//    Authorization: `Bearer ${auth}`,
//  };
//
//  const request = new Request(url, {
//    method: 'DELETE',
//    headers: headers,
//  });
//
//  const response = await fetch(request);
//  const data = await response;
//  // Handle the response as needed
//};
//
//export { deleteMember };
//
//
//const addMember = async (auth, wid, mid) => {
//  console.log("Add Member")
//  console.log(mid)
//  const url = 'https://cse118.com/api/v2/workspace/${wid}/member?mid=${mid}`;
//  console.log("URL")
//  console.log(url)
//
// // const midQueryItem = new URLSearchParams({ mid: mid });
////  url.search = midQueryItem.toString();
//
//  const headers = {
//    Accept: 'application/json',
//    Authorization: `Bearer ${auth}`,
//  };
//
//  const response = await fetch(url, {
//    method: 'POST',
//    headers: headers,
//  });
//  const data = await response.text();
//  return data;
//  // Handle the response data as needed
//};
//
//export { addMember };

File: Assignment6-1.Submission/Assignment6/Repo/MemberRepo.swift
//
//  MemberRepo.swift
//  Assignment6
//
//  Created by Abhinav Sunil on 5/16/23.
//

import Foundation

struct MemberRepo {
    static func load(auth: String) async throws -> [OtherMember] {
      //  print("Auth")
     //   print(auth)
       // sleep(1) 
        let url = URL(string: "https://cse118.com/api/v2/member")!
        var request = URLRequest(url: url)
        request.setValue("application/json", forHTTPHeaderField: "Accept") // Set the Accept header
        request.setValue("Bearer \(auth)", forHTTPHeaderField: "Authorization") // Set the Authorization header

        let (data, _) = try await URLSession.shared.data(for: request)// Use the prepared request
//        if let jsonString = String(data: data, encoding: .utf8) {
//          //  print(jsonString)
//        }
        
        
        return try JSONDecoder().decode([OtherMember].self, from: data)
    }
    static func get(auth: String, id: String) async throws -> [OtherMember]{
      //  print("Auth")
     //   print(auth)
       // sleep(1)
        let url = URL(string: "https://cse118.com/api/v2/workspace/\(id)/member")!
        var request = URLRequest(url: url)
        request.setValue("application/json", forHTTPHeaderField: "Accept") // Set the Accept header
        request.setValue("Bearer \(auth)", forHTTPHeaderField: "Authorization") // Set the Authorization header

        let (data, _) = try await URLSession.shared.data(for: request) // Use the prepared request
        
//        if let jsonString = String(data: data, encoding: .utf8) {
//          //  print(jsonString)
//        }
        
        return try JSONDecoder().decode([OtherMember].self, from: data)
    }
    
    static func delete(auth: String, wid: String, mid: String) async throws {
        let urlString = "https://cse118.com/api/v2/workspace/\(wid)/member"
        var urlComponents = URLComponents(string: urlString)
        
        let midQueryItem = URLQueryItem(name: "mid", value: mid)
        urlComponents?.queryItems = [midQueryItem]
        
        let url = urlComponents?.url
        var request = URLRequest(url: url!)
        
        
        request.setValue("application/json", forHTTPHeaderField: "Accept") // Set the Accept header
        request.setValue("Bearer \(auth)", forHTTPHeaderField: "Authorization")
        request.httpMethod = "DELETE"
        
        let (data, _) = try await URLSession.shared.data(for: request)
        
    }
    
    static func add(auth: String, wid: String, mid: String) async throws {
        let urlString = "https://cse118.com/api/v2/workspace/\(wid)/member"
        var urlComponents = URLComponents(string: urlString)
        
        let midQueryItem = URLQueryItem(name: "mid", value: mid)
        urlComponents?.queryItems = [midQueryItem]
        
        let url = urlComponents?.url
        
        var request = URLRequest(url: url!)
        request.setValue("application/json", forHTTPHeaderField: "Accept")
        request.setValue("Bearer \(auth)", forHTTPHeaderField: "Authorization")
        request.httpMethod = "POST"
        
        let (data, _) = try await URLSession.shared.data(for: request)
        // Handle the response data as needed
    }
}
